float64 puck_offset
float64 puck_distance
